{"uid":"8e102944446fb09e","name":"Проверка невозможности создания билд конфигурации","fullName":"tests.backend.test_create_build_conf.TestBuildConfCreate#test_create_build_conf_with_empty_attributes","historyId":"4ba5ecaf909ddf5321128bd1db749264","time":{"start":1711452159768,"stop":1711452159768,"duration":0},"description":"Проверяется валидация с пустыми атрибутами билд конфигурации","descriptionHtml":"<p>Проверяется валидация с пустыми атрибутами билд конфигурации</p>\n","status":"broken","statusMessage":"ValueError: Unexpected status code: 401, expected: 200","statusTrace":"self = <api.auth_api.AuthAPI object at 0x7f75987cbe50>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n>           csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n\napi/auth_api.py:9: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f75987cbe50>, method = 'GET'\nendpoint = '/authenticationTest.html?csrf', data = None\nexpected_status = <HTTPStatus.OK: 200>, need_logging = True\nneed_verify_status = True\n\n    def send_request(self, method, endpoint, data=None, expected_status=HTTPStatus.OK, need_logging=True,\n                     need_verify_status=True):\n        \"\"\"\n        Врапер для запросов. Позволяет прикручивать дополнительную логику\n        :param method: Метод запроса\n        :param endpoint: Ендпоинт для склейки с BASE_URL\n        :param data: Тело запроса. По умолчанию пустое, чтобы пропускало no-content запросы\n        :param expected_status: Ожидаемый статус код\n        :param need_logging: Флаг для логирования. По умолчанию = True\n        :param need_verify_status: Флаг для проверки статус кода в ответе. По умолчанию = True\n        :return: Возвращает объект ответа\n        \"\"\"\n        url = f\"{self.base_url}{endpoint}\"\n        response = self.session.request(method=method, url=url, json=data)\n        if need_logging:\n            self.log_request_and_response(response, expected_status, need_verify_status)\n        if need_verify_status:\n            if response.status_code != expected_status:\n>               raise ValueError(f\"Unexpected status code: {response.status_code}, expected: {expected_status}\")\nE               ValueError: Unexpected status code: 401, expected: 200\n\ncustom_requester/custom_requester.py:34: ValueError\n\nDuring handling of the above exception, another exception occurred:\n\nuser_session = <function user_session.<locals>._create_user_session at 0x7f75987bc4c0>\n\n    @pytest.fixture\n    def super_admin(user_session):\n        new_session = user_session()\n        super_admin = User(SuperAdminCreds.USERNAME, SuperAdminCreds.PASSWORD, None, new_session,\n                           [\"SUPER_ADMIN\"])\n>       super_admin.api_manager.auth_api.auth_and_get_csrf(super_admin.creds)\n\nconftest.py:51: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f75987cbe50>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n            csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n        except Exception as e:\n>           raise ValueError(f\"{e}\")\nE           ValueError: Unexpected status code: 401, expected: 200\n\napi/auth_api.py:11: ValueError","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1711452159621,"stop":1711452159658,"duration":37},"status":"passed","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"hasContent":false,"attachmentStep":false,"shouldDisplayMessage":false,"stepsCount":0},{"name":"super_admin","time":{"start":1711452159769,"stop":1711452159776,"duration":7},"status":"broken","statusMessage":"ValueError: Unexpected status code: 401, expected: 200\n","statusTrace":"  File \"/home/runner/.local/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/.local/lib/python3.10/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n  File \"/home/runner/.local/lib/python3.10/site-packages/pluggy/_callers.py\", line 102, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/.local/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1149, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/.local/lib/python3.10/site-packages/_pytest/fixtures.py\", line 923, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/TeamCity_Test/TeamCity_Test/conftest.py\", line 51, in super_admin\n    super_admin.api_manager.auth_api.auth_and_get_csrf(super_admin.creds)\n  File \"/home/runner/work/TeamCity_Test/TeamCity_Test/api/auth_api.py\", line 11, in auth_and_get_csrf\n    raise ValueError(f\"{e}\")\n","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"hasContent":true,"attachmentStep":false,"shouldDisplayMessage":true,"stepsCount":0},{"name":"user_session","time":{"start":1711452159769,"stop":1711452159769,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"hasContent":false,"attachmentStep":false,"shouldDisplayMessage":false,"stepsCount":0}],"testStage":{"description":"Проверяется валидация с пустыми атрибутами билд конфигурации","status":"broken","statusMessage":"ValueError: Unexpected status code: 401, expected: 200","statusTrace":"self = <api.auth_api.AuthAPI object at 0x7f75987cbe50>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n>           csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n\napi/auth_api.py:9: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f75987cbe50>, method = 'GET'\nendpoint = '/authenticationTest.html?csrf', data = None\nexpected_status = <HTTPStatus.OK: 200>, need_logging = True\nneed_verify_status = True\n\n    def send_request(self, method, endpoint, data=None, expected_status=HTTPStatus.OK, need_logging=True,\n                     need_verify_status=True):\n        \"\"\"\n        Врапер для запросов. Позволяет прикручивать дополнительную логику\n        :param method: Метод запроса\n        :param endpoint: Ендпоинт для склейки с BASE_URL\n        :param data: Тело запроса. По умолчанию пустое, чтобы пропускало no-content запросы\n        :param expected_status: Ожидаемый статус код\n        :param need_logging: Флаг для логирования. По умолчанию = True\n        :param need_verify_status: Флаг для проверки статус кода в ответе. По умолчанию = True\n        :return: Возвращает объект ответа\n        \"\"\"\n        url = f\"{self.base_url}{endpoint}\"\n        response = self.session.request(method=method, url=url, json=data)\n        if need_logging:\n            self.log_request_and_response(response, expected_status, need_verify_status)\n        if need_verify_status:\n            if response.status_code != expected_status:\n>               raise ValueError(f\"Unexpected status code: {response.status_code}, expected: {expected_status}\")\nE               ValueError: Unexpected status code: 401, expected: 200\n\ncustom_requester/custom_requester.py:34: ValueError\n\nDuring handling of the above exception, another exception occurred:\n\nuser_session = <function user_session.<locals>._create_user_session at 0x7f75987bc4c0>\n\n    @pytest.fixture\n    def super_admin(user_session):\n        new_session = user_session()\n        super_admin = User(SuperAdminCreds.USERNAME, SuperAdminCreds.PASSWORD, None, new_session,\n                           [\"SUPER_ADMIN\"])\n>       super_admin.api_manager.auth_api.auth_and_get_csrf(super_admin.creds)\n\nconftest.py:51: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f75987cbe50>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n            csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n        except Exception as e:\n>           raise ValueError(f\"{e}\")\nE           ValueError: Unexpected status code: 401, expected: 200\n\napi/auth_api.py:11: ValueError","steps":[],"attachments":[{"uid":"f3e2bd2a46b9916a","name":"log","source":"f3e2bd2a46b9916a.txt","type":"text/plain","size":568}],"parameters":[],"attachmentsCount":1,"hasContent":true,"attachmentStep":false,"shouldDisplayMessage":true,"stepsCount":0},"afterStages":[{"name":"user_session::0","time":{"start":1711452159801,"stop":1711452159802,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"hasContent":false,"attachmentStep":false,"shouldDisplayMessage":false,"stepsCount":0}],"labels":[{"name":"suite","value":"Управление билд конфигурациями"},{"name":"severity","value":"normal"},{"name":"subSuite","value":"Негативные сценарии"},{"name":"parentSuite","value":"API тесты"},{"name":"tag","value":"api"},{"name":"tag","value":"build_conf"},{"name":"host","value":"fv-az1391-496"},{"name":"thread","value":"7848-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.backend.test_create_build_conf"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"attribute","value":"'project_id'"},{"name":"status_code","value":"<HTTPStatus.NOT_FOUND: 404>"},{"name":"text_error","value":"'NotFoundException: No project found by locator 'count:1,id:'"},{"name":"value","value":"''"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":1,"skipped":0,"passed":3,"unknown":0,"total":4},"items":[{"uid":"41e77d7a8ebd7f33","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/121//#testresult/41e77d7a8ebd7f33","status":"passed","time":{"start":1711451062580,"stop":1711451062690,"duration":110}},{"uid":"4653af6b3e9c841b","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/120//#testresult/4653af6b3e9c841b","status":"passed","time":{"start":1711369183380,"stop":1711369183497,"duration":117}},{"uid":"645a72ffe648eff3","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/119//#testresult/645a72ffe648eff3","status":"passed","time":{"start":1711356084294,"stop":1711356084411,"duration":117}}]},"tags":["api","build_conf"]},"source":"8e102944446fb09e.json","parameterValues":["'project_id'","<HTTPStatus.NOT_FOUND: 404>","'NotFoundException: No project found by locator 'count:1,id:'","''"]}